### PR TITLE
[CU-3cq0h2w] feat: put Report

### PR DESCRIPTION
**Pull request by RonnyChiang**

### CODE REVIEW

**Comment by WendellLiu** [comment]
Task linked: [CU-3cq0h2w PUT /admin/photos/:id/resolve](https://app.clickup.com/t/3cq0h2w)

**Comment by RonnyChiang** [comment]
[API doc](https://app.clickup.com/25699357/v/dc/rg90x-245/rg90x-265?block=block-db380d9b-4c52-429c-b944-5d6eb9774e2b)

File: src/services/report-service/report-service.test.js [hunk 1]
```diff
@@ -312,3 +313,291 @@ describe('report-service.postReport', () => {
...
+        async () => {
+          await reportService.putReportsByReportId(report.id, 'no_issue');
+        },
+        {
+          code: reportErrorMap.reportAlreadyResolve.code,
+        }
+      );
+    });
+  });
+  describe('with not exist reportId', () => {
```
Comment (WendellLiu) [inline src/services/report-service/report-service.test.js:530]: with no or without

File: src/services/report-service/index.js [hunk 2]
```diff
@@ -88,6 +89,26 @@ class ReportService extends BaseService {
...
+    if (!hasReport) throw new GeneralError(reportErrorMap['wrongReportId']);
+    if (hasReport && hasReport.status !== 'pending')
+      throw new GeneralError(reportErrorMap['reportAlreadyResolve']);
+
+    // put reports by photoId
+    const photoId = hasReport.photoId;
+    const putReports = await reportModel.putReportsByPhotoId(photoId, status);
+
+    // delete photo if status is deleted
+    if (status === 'deleted') {
```
Comment (WendellLiu) [inline src/services/report-service/index.js:105]: status map

**Review by WendellLiu — CHANGES_REQUESTED** [review]

File: src/models/report/index.js [hunk 3]
```diff
@@ -67,6 +67,34 @@ class ReportModel {
...
+        userId: true,
+        photoId: true,
+        status: true,
+        content: true,
+        createdAt: true,
+      },
+    });
+    return report;
+  }
+  async putReportsByPhotoId(photoId, status) {
```
[the remark this answers is not in the document]
Comment (RonnyChiang) [reply src/models/report/index.js]: fixed

File: src/services/photo-service/index.js [hunk 4]
```diff
@@ -116,16 +116,16 @@ class PhotoService extends BaseService {
       //get start photo
       const startPhoto = await photoModel.getPhoto(startPhotoId);
 
-      if (isNil(startPhoto))
+      if (isEmpty(startPhoto))
```
[the remark this answers is not in the document]
Comment (RonnyChiang) [reply src/services/photo-service/index.js]: 修正

File: src/services/report-service/index.js [hunk 5]
```diff
@@ -88,6 +89,26 @@ class ReportService extends BaseService {
     };
     return result;
   }
+  async putReportsByReportId(reportId, status) {
+    // check report exist and status is pending
+    const reportModel = new ReportModel();
+    const hasReport = await reportModel.getReportByReportId(parseInt(reportId));
```
[the remark this answers is not in the document]
Comment (RonnyChiang) [reply src/services/report-service/index.js]: fixed

File: src/services/report-service/report-service.test.js (hunk 1 above)
Comment (RonnyChiang) [reply src/services/report-service/report-service.test.js]: fixed

File: src/services/report-service/index.js (hunk 2 above)
Comment (RonnyChiang) [reply src/services/report-service/index.js]: fixed

File: src/models/photo/index.js [hunk 6]
```diff
@@ -154,6 +164,23 @@ class PhotoModel {
 
     return photosWithReviewCount;
   }
+  async deletePhoto(photoId) {
+    const deletePhoto = await prisma.photos.update({
+      where: { id: photoId },
+      data: {
+        isDeleted: 1,
+        deletedAt: new Date(Date.now()),
```
Comment (WendellLiu) [inline src/models/photo/index.js:172]: 這跟 new Date() 有什麼不同嗎？

File: src/services/report-service/index.js [hunk 7]
```diff
@@ -88,6 +89,26 @@ class ReportService extends BaseService {
     };
     return result;
   }
+  async putReportsByReportId(reportId, status) {
+    // check report exist and status is pending
+    const reportModel = new ReportModel();
+    const report = await reportModel.getReportByReportId(parseInt(reportId));
+    if (!report) throw new GeneralError(reportErrorMap['wrongReportId']);
+    if (report && report.status !== 'pending')
```
Comment (WendellLiu) [inline src/services/report-service/index.js:97]: status map

**Review by WendellLiu — COMMENTED** [review]
主要邏輯沒問題，幾個小地方

File: src/models/photo/index.js (hunk 6 above)
Comment (RonnyChiang) [reply src/models/photo/index.js]: 沒有，已修正

File: src/services/report-service/index.js (hunk 7 above)
Comment (RonnyChiang) [reply src/services/report-service/index.js]: fixed

**Review by WendellLiu — APPROVED** [review]

**Pull request merged**